pkg/controller/controller.go

```diff
@@ -199,6 +204,12 @@ func (c *Controller) scaleNodeGroup(nodegroup string, nodeGroup *NodeGroupState)
 		return 0, err
 	}
 
+	// store a cached version of node capacity
+	if (nodeGroup.cpuCapacity.IsZero() || nodeGroup.memCapacity.IsZero()) && len(allNodes) > 0 {
```

Comment:
yeah, thought about that when I updated controller. The thing is that it's against escalator assumption inside design document which is `All nodes in the node group have the same allocatable resources`. But at the same time, to avoid restarting escalator after that change happens in one of the node group, we could update that value in each run which will automatically fix itself during the first run after all nodes inside node group have been updated. I'll remove the value check for those cached capacity.